### PR TITLE
Log debugging info instead of printing it

### PR DESCRIPTION
**Pull request by djmitche**
Logging doesn't appear in the summary, but is available afterward.
Printing appears when the test is run, which may be many lines above the
summary.

This also adjusts the assertion messages to include information about
the received value, rather than just stating that it differs from the
expected value.

### CODE REVIEW

File: master/buildbot/test/util/steps.py [hunk 1]
```diff
@@ -335,7 +356,7 @@ def check(result):
 
     @defer.inlineCallbacks
     def _validate_expectation(self, exp, command):
-        got = (command.remote_command, command.args)
+        got=(command.remote_command, command.args)
```
Comment (tardyp) [inline master/buildbot/test/util/steps.py]: pep8?

File: master/buildbot/test/util/steps.py [hunk 2]
```diff
@@ -51,9 +51,12 @@ def _dict_diff(d1, d2):
 
     for k in both:
         if isinstance(d1[k], dict) and isinstance(d2[k], dict):
-            missing_in_v1, missing_in_v2, different_in_v = _dict_diff(d1[k], d2[k])
-            missing_in_d1.extend(['{0}.{1}'.format(k, m) for m in missing_in_v1])
-            missing_in_d2.extend(['{0}.{1}'.format(k, m) for m in missing_in_v2])
+            missing_in_v1, missing_in_v2, different_in_v = _dict_diff(
```
Comment (tardyp) [inline master/buildbot/test/util/steps.py]: not sure if those are really needed. I usually cut at 100. cutting at 72 will mean a lot of such refacto, and to me loss of readability.

**Comment by tardyp** [comment]
some nits. obviously :+1: on the principle

File: master/buildbot/test/util/steps.py (hunk 2 above)
Comment (djmitche) [inline master/buildbot/test/util/steps.py]: I think autopep8 did it (and the `got=(..` below)

Comment (tardyp) [inline master/buildbot/test/util/steps.py]: mmhh very weird. I think you have a wrong autopep8 version or badly configured editor?

the autopep8 from the hooks is enforcing 100 columns, not 72.
I just verified that I can modify that file, and autopep8 wont remove the spaces between = nor reduce lines

**Comment by tardyp** [comment]
:+1: with the new version

**Pull request merged**